### PR TITLE
Better upgrade tests to increase confidence about migrations

### PR DESCRIPTION
**Pull request by albcui**

### CODE REVIEW

File: src/testing/deploy/upgrade_test.go [hunk 1]
```diff
@@ -261,6 +262,88 @@ func TestUpgradeOpenCVWithAuth(t *testing.T) {
 	)
 }
 
+func TestUpgradeMultiProjectJoins(t *testing.T) {
+	if skip {
+		t.Skip("Skipping upgrade test")
+	}
+	fromVersions := []string{"2.5.0", "2.6.0"}
```
Comment (jrockway) [inline src/testing/deploy/upgrade_test.go:269]: We should probably include 2.4 here as well.

**Review by jrockway — APPROVED** [review]
This looks good.

File: src/testing/deploy/upgrade_test.go (hunk 1 above)
Comment (albscui) [reply src/testing/deploy/upgrade_test.go:269]: So creation of projects is a new feature introduced in 2.5, and our auth interceptor expects it to be auth gated. Our test creates projects, so it's fundamentally incompatible with 2.4.

Comment (jrockway) [reply src/testing/deploy/upgrade_test.go:269]: ahh, ok!

**Pull request merged**